src/machine/machine-varlink.c

```diff
@@ -0,0 +1,156 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "hostname-util.h"
+#include "json.h"
+#include "machine-varlink.h"
+#include "machine.h"
+#include "path-util.h"
+#include "pidref.h"
+#include "process-util.h"
+#include "sd-id128.h"
+#include "socket-util.h"
+#include "string-util.h"
+#include "varlink.h"
+#include <limits.h>
+
+static JSON_DISPATCH_ENUM_DEFINE(dispatch_machine_class, MachineClass, machine_class_from_string)
+
+static int machine_name(const char *name, JsonVariant *variant, JsonDispatchFlags flags, void *userdata) {
+        char **m = ASSERT_PTR(userdata);
+        int r;
+
+        assert(variant);
+
+        if (!json_variant_is_string(variant))
+                return json_log(variant, flags, SYNTHETIC_ERRNO(EINVAL), "JSON field '%s' is not a string.", strna(name));
+
+        if (!hostname_is_valid(json_variant_string(variant), /* flags= */ 0))
```

Comment:
even though json_variant_string() should be quite quick i#d still store the result in a local variable, given you are going to use it below a 2nd time